Keep original network

jamesallardice opened the pull request.

jamesallardice committed:
Add original network to Network instance

This will allow consumers to access the original network on the parsed
instance.

tyom commented on src/match.ts:
```diff
@@ -36,16 +36,16 @@ export class Matcher {
    * @param network - An address or subnet
    *
-   * @returns A boolean
+   * @returns A Network or null
```
It's a bit odd that the method named `has` returns a non-boolean value. Makes it less intuitive as generally a method with that name would be expected to return a boolean.

Could it be a new method instead? For example, `getNetwork`.

jamesallardice replied:
Yeah agreed. Have updated this.

jamesallardice committed:
Add method to return matched network

This will allow consumers to access the original network on the parsed
instance.

tyom commented:
Perhaps take an opportunity to fix vulnerabilities found by npm audit?

**Before `npm audit fix`**
![image](https://user-images.githubusercontent.com/203886/90004075-f27c8380-dc8c-11ea-8614-3fe9baba75c7.png)

**After**
![image](https://user-images.githubusercontent.com/203886/90004111-02946300-dc8d-11ea-966d-2450262bda97.png)

tyom commented on src/match.ts:
```diff
@@ -48,6 +48,28 @@ export class Matcher {
+   * @example
+   * var matcher = new netparser.Matcher();
+   * matcher.add("10.0.0.0/24");
+   * matcher.has("192.168.5.0"); // returns null
```
Should update the method name in example.

jamesallardice committed:
Update deps via npm audit fix

markwallsgrove commented on src/network.ts:
```diff
@@ -8,9 +8,12 @@ const AFTER = 1;
   private netbits = -1;
 
   public constructor(network?: string, throwErrors?: boolean) {
+    this.network = network;
```
I guess they don't do this to avoid memory bloat. What does the memory look like if we load all ranges into memory? I guess it isn't too bad?

jamesallardice replied:
I haven't checked but in our case the ranges are ~3000 strings, all of which will already be in memory because we load in the whole lot and parse it as JSON. I don't expect this to cause any problems.
